daemon/daemon_linux.go

```diff
@@ -33,11 +33,11 @@ func (daemon *Daemon) cleanupMountsFromReader(reader io.Reader, unmount func(tar
 	for sc.Scan() {
 		line := sc.Text()
 		fields := strings.Fields(line)
-		if strings.HasPrefix(fields[4], daemon.repository) {
-			logrus.Debugf("Mount base: %v, repository %s", fields[4], daemon.repository)
+		if strings.HasPrefix(fields[4], daemon.root) {
+			logrus.Debugf("Mount base: %v", fields[4])
 			mnt := fields[4]
 			mountBase := filepath.Base(mnt)
-			if mountBase == "mqueue" || mountBase == "shm" {
+			if mountBase == "mqueue" || mountBase == "shm" || mountBase == "merged" {
```

Comment:
I'm not sure I like this.
Cleanup is tied to overlay. Wouldn't other graphdrivers have this issue as well?